pkg/appmanager/resourceConfig.go

```diff
@@ -1742,16 +1740,41 @@ func (appMgr *Manager) createRSConfigFromRoute(
 
 // Copies from an existing config into our new config
 func (rc *ResourceConfig) copyConfig(cfg *ResourceConfig) {
-	var cfgBytes bytes.Buffer
-	err := gob.NewEncoder(&cfgBytes).Encode(cfg)
-	if err != nil {
-		log.Errorf("Couldn't encode ResourceConfig: %v", err)
-		return
-	}
-	err = gob.NewDecoder(&cfgBytes).Decode(&rc)
-	if err != nil {
-		log.Errorf("Could not decode ResourceConfig: %v", err)
-		return
+	// MetaData
+	rc.MetaData = cfg.MetaData
+	// Virtual
+	rc.Virtual = cfg.Virtual
+	// Profiles
+	rc.Virtual.Profiles = make([]ProfileRef, len(cfg.Virtual.Profiles))
```

Comment:
I don't see rc.Virtual.Policies or rc.Virtual.IRules being copied over like you are doing for rc.Virtual.Profiles.